internal/directory/okta/okta.go

```diff
@@ -145,7 +157,51 @@ func (p *Provider) UserGroups(ctx context.Context) ([]*directory.Group, []*direc
 	return groups, users, nil
 }
 
+func (p *Provider) getUpdatedGroup(ctx context.Context) ([]*directory.Group, error) {
+	now := time.Now()
+	if p.lastUpdated == nil {
+		p.lastUpdated = &now
+	}
+
+	u := &url.URL{Path: "/api/v1/groups"}
+	q := u.Query()
+	q.Set("limit", strconv.Itoa(p.cfg.batchSize))
+	q.Set("filter", fmt.Sprintf(`lastUpdated+gt+"%[1]s"+or+lastMembershipUpdated+gt+"%[1]s"`, p.lastUpdated.UTC().Format(filterDateFormat)))
+	u.RawQuery = q.Encode()
+	p.lastUpdated = &now
```

Comment:
This may be ok as is, but sometimes clocks are out of sync and it's generally better to rely on whatever timestamps the server returns. For example, we could loop over all the returned records and use the latest `lastUpdated` or `lastMembershipUpdated` timestamp.

Reply:
👍